File: Python Fundamentals/Mid Exam problems/computer_store.py
total_price = 0
total_taxes = 0
total = 0
while True:

    inp = input()

    if inp == 'special' or inp == 'regular':
        break

    price = float(inp)
    if price < 0:
        print('Invalid price!')
        continue

    tax = 0.20 * price
    total_taxes += tax
    total_price += price

total = total_price + total_taxes
if inp == 'special':
    total -= 0.10 * total

if total == 0:
    print("Invalid order!")
else:
    print("Congratulations you've just bought a new computer!")
    print(f'Price without taxes: {total_price:.2f}$')
    print(f'Taxes: {total_taxes:.2f}$')
    print('-----------')
    print(f'Total price: {total:.2f}$')

File: JavaScript Fundamentals/Mid Exams/Mid Exam 1/computerStore.js
function solve(arr) {
    let discount = false;
    let totalPrice = 0;
    let taxes = 0;
    let priceWithoutTaxes = 0;

    for (let el of arr) {
        if (el == 'regular') break;
        else if (el == 'special') discount = true;
        else if (Number(el) < 0) console.log("Invalid price!");
        else {
        totalPrice += Number(el);
        taxes += 0.20 * Number(el);
        }
    }    
    priceWithoutTaxes = totalPrice;
    totalPrice += taxes;
    if (discount) totalPrice -= 0.10 * totalPrice;
    if (totalPrice == 0) console.log("Invalid order!");
    else {
        console.log("Congratulations you've just bought a new computer!");
        console.log(`Price without taxes: ${priceWithoutTaxes.toFixed(2)}$`);
        console.log(`Taxes: ${taxes.toFixed(2)}$`);
        console.log('-----------');
        console.log(`Total price: ${totalPrice.toFixed(2)}$`);
    }
}
